udata/core/user/models.py

```diff
@@ -210,6 +218,33 @@ def json_ld(self):
 
         return result
 
+    def mark_as_deleted(self):
+        copied_user = copy(self)
+        self.email = 'Deleted-{}'.format(self.id)
+        self.password = None
+        self.active = False
+        self.first_name = 'Deleted'
+        self.last_name = 'Deleted'
+        self.avatar = None
+        self.avatar_url = None
+        self.website = None
+        self.about = None
+        self.deleted = datetime.now()
+        self.save()
+        for organization in self.organizations:
+            organization.members = [member
+                                    for member in organization.members
+                                    if member.user != self]
+            organization.save()
+        for discussion in self.involving_discussions:
+            for message in discussion.discussion:
+                if message.posted_by == self:
+                    message.content = 'Deleted'
```

Comment:
Yes, it was discussed before starting this story. But I agree, just putting 'Deleted' is a little bit hardcore (even more for french based portals). Maybe we can add a deleted flag and display deleted in the front.